### ISSUE TITLE
Request: ngModel/ng-change support  or an way to specify an specific input-model attribute for output-model

### ISSUE THREAD

**Issue by stefanvitz** [isteven/angular-multi-select#231]
Hi isteven,

I just tried your multi-select directive and is really works fine (something that is not true for most 'premade'-angular directives).
Only thing i noticed is that it (because of the input-/output-model thing) does not support ngModel and thereby no binding of ng-change.
So when i whant to call an change-callback i have to register multiple callbacks e.g. onClick, onReset, onSelectAll, etc. wich is not very programmer-friendly.

what i need is an option to specify which field of the input-data element to store in the output-data (e.g. it's id/key as in ng-options obj.key as ob.label for options)  or an hook to manually process the output model. e.g. extract the ids of the selected elements (e.g ng-change).
i would like to avoid $watch due to an dynamic amount of multi-selects and performance-considerations. 

regards Stefan

**Comment by stefanvitz** [comment]
i'm unable to actually do an commit and create an pull-request but currently i?m working with the following patch:

```javascript
Index: dev/src/main/resources/META-INF/resources/lib/multi-select/isteven-multi-select-3.0.0.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dev/src/main/resources/META-INF/resources/lib/multi-select/isteven-multi-select-3.0.0.js	(revision Local version)
+++ dev/src/main/resources/META-INF/resources/lib/multi-select/isteven-multi-select-3.0.0.js	(revision Shelved version)
@@ -43,6 +43,7 @@
             // models
             inputModel      : '=',
             outputModel     : '=',
+            outputAttribute : '@',
 
             // settings based on attribute
             buttonLabel     : '@',
@@ -498,8 +499,13 @@
                     if ( typeof value !== 'undefined' ) {
                         if ( typeof value[ $scope.groupProperty ] === 'undefined' ) {
                             if ( value[ $scope.tickProperty ] === true ) {
-                                // selectedItems.push( value );
-                                var temp = angular.copy( value );
+                                var temp;
+                                if ($scope.outputAttribute == undefined) {
+                                    temp = angular.copy(value );
+                                }
+                                else {
+                                    temp = angular.copy(value[$scope.outputAttribute]);
+                                }
                                 var index = $scope.outputModel.push( temp );
                                 delete $scope.outputModel[ index - 1 ][ $scope.indexProperty ];
                                 delete $scope.outputModel[ index - 1 ][ $scope.spacingProperty ];
@@ -534,8 +540,8 @@
                         $scope.more = false;
                     }
 
-                    angular.forEach( $scope.outputModel, function( value, key ) {
-                        if ( typeof value !== 'undefined' ) {
+                    angular.forEach( $scope.localModel, function( value, key ) {
+                        if ( typeof value !== 'undefined' && value[ $scope.tickProperty ] === true ) {
                             if ( ctr < tempMaxLabels ) {
                                 $scope.varButtonLabel += ( $scope.varButtonLabel.length > 0 ? '</div>, <div class="buttonLabel">' : '<div class="buttonLabel">') + $scope.writeLabel( value, 'buttonLabel' );
                             }
```

**Comment by isteven** [comment]
Hi @stefanvitz,

If you read AngularJs <code>ng-change</code> manual (https://docs.angularjs.org/api/ng/directive/ngChange), you'll see that <code>ng-change</code> is intended for a single input element, while this multi-select is basically a logic play on the array of objects (there are checkboxes input elements, but only to accept keyboard inputs). I can easily change the attribute name from "input-model" to "ng-model", but I doubt it will do anything good. In short, I don't think ng-change is the solution here.

To specify the output-model properties, use PR https://github.com/isteven/angular-multi-select/pull/127 - You need to modify things a bit.. I'm quite occupied at the moment so won't be able to touch the directive for a while.

Cheers.

**Comment by isteven** [comment]
output-properties is now available in v4.0.0
